### ISSUE TITLE
Transforming Phlex code output through RuboCop or SyntaxTree

### ISSUE THREAD

**Issue by marcoroth** [marcoroth/phlexing#59]
Let's say we have this:
```erb
<%= tag.div class: something? ? "class-1" : "class-2" do %>
  <%= content_tag :span, @something.text %>
<% end %>
```
Which would generate this through phlexing today:
```ruby
class SomethingComponent < Phlex::HTML
  attr_accessor :something

  def initialize(something:)
    @something = something
  end

  def template
    tag.div class: something? ? "class-1" : "class-2" do
      content_tag :span, something.text
    end
  end
end
```
But since this is a regular Ruby class now we could write any RuboCop rule (or maybe even a SyntaxTree mutation visitor) which statically analyses the code and auto-fixes them (we could even make the rules toggleable from the UI)

For example some rules could be:
* rewrite `tag.div` as `div { ... }`
* rewrite `content_tag :span` to `span { ... }`
* rewrite all references of `something` inside `def template` to `@something` and remove the `attr_accessor` for it
* rewrite `class: something? ? "class-1" : "class-2"` to `**classes(something?: { then: "class-1", else: "class-2" })`
* ...

We could even release these rules standalone as `rubocop-phlex` which people could use in their project independent of phlexing

**Comment by joeldrapper** [comment]
A SyntaxTree mutation visitor would probably be the easiest solution for this.

**Comment by jaredcwhite** [comment]
(Could be tricky with things like lists, where a single `<template>` tag with directives in its content would need to be duplicated and expanded out for n items in an array.)

**Comment by marcoroth** [comment]
@jaredcwhite maybe I'm not understanding 100% what you are trying to do, but it seems kinda overkill to me to use phlexing (or phlex for that matter) as an intermediate format for those kind of transformations. Unless you already have those transformations written and implemented based on a Ruby AST.

**Comment by joeldrapper** [comment]
I agree. I think you’ll find it's easier to use Nokogiri than SyntaxTree for this kind of thing @jaredcwhite.

**Comment by jaredcwhite** [comment]
> If PhlexML is the final target, you can always do the transformations in Nokogiri first, then run it through Phlexing.

Interesting…I could essentially inject ERB into the markup via Nokogiri prior to the Phlexing stage. I'll give that a try!

**Comment by marcoroth** [comment]
@jaredcwhite depening what you are doing to might also find `Phlexing::ERBTransformer` or `Phlexing::Parser` helpful so you don't need to replace the output tags with weird markers
